tests/api_fastapi/core_api/routes/public/test_dag_run.py

```diff
@@ -931,6 +932,13 @@ def test_patch_dag_run(self, test_client, dag_id, run_id, patch_body, response_b
         assert body["dag_run_id"] == run_id
         assert body.get("state") == response_body.get("state")
         assert body.get("note") == response_body.get("note")
+        user_id = (
+            session.query(DagRunNote)
+            .join(DagRun, DagRunNote.dag_run_id == DagRun.id)
```

Comment:
For some reason, this query is failing with
 `AttributeError: 'DagRunNote' object has no attribute 'dag_id'`

Even a simple `session.query(DagRunNote).all()` is failing with the above error. Any idea what's going on?

Reply:
I see that the error is coming from DagRunNote's `__repr__` method:

```
    def __repr__(self):
        prefix = f"<{self.__class__.__name__}: {self.dag_id}.{self.dagrun_id} {self.run_id}"
        if self.map_index != -1:
            prefix += f" map_index={self.map_index}"
        return prefix + ">"
```

DagRunNote does not have any of dag_id, dagrun_id and run_id. Not sure why we added these here.

CC: @pierrejeambrun

Reply:
however, if I directly access the field like below:

```
result = session.query(DagRunNote).join(DagRun, DagRunNote.dag_run_id==DagRun.id).filter(DagRun.run_id == run_id).one()

result.user_id
```

It does not fail and it does print the user_id. So, the issue is definitely with the __repr__ method trying to access unavailable fields

Reply:
Definitely not related to this issue. But, probably can be a follow up PR to fix this, if this is a bug

Reply:
Yep, strange piece of code. Might be worth to take a look and fix in a separate PR.

Reply:
In query context it most likely work because of the implicit join on the `DagRun` table via the relationship.